srcCpp/Infrastructure_pro.cxx

```diff
@@ -822,6 +833,26 @@ bool PerftestConfigureTransport(
 
         case TRANSPORT_SHMEM:
             qos.transport_builtin.mask = DDS_TRANSPORTBUILTIN_SHMEM;
+            break;
+
+        default:
+            break;
+    };
+
+    /*
+     * Once the configurations have been stablished, we can get the
+     * MessageSizeMax for the Transport, which should be the minimum of
+     * all the enabled transports
+     */
+    configureMessageSizeMaxTransport(transport, qos);
+
+    switch (transport.transportConfig.kind) {
+
+        case TRANSPORT_UDPv4:
+        case TRANSPORT_UDPv6:
+            break;
+
+        case TRANSPORT_SHMEM:
             if (!configureShmemTransport(transport, qos, _PM)) {
```

Comment:
Should we maybe do this only if the 2 properties are not set via code? Like:

if (DDSPropertyQosPolicyHelper::lookup_property(qos.property,
                    "dds.transport.shmem.builtin.parent.received_message_count_max") == NULL
            && DDSPropertyQosPolicyHelper::lookup_property(qos.property,
                    "dds.transport.shmem.builtin.parent.receive_buffer_size") == NULL) {
    if (!configureShmemTransport(transport, qos, _PM)) {
                fprintf(stderr,
                        "%s Failed to configure SHMEM plugin\n",
                        classLoggingString.c_str());
        return false;
    }
}
break;